CONTRIBUTIONS.md

```diff
@@ -0,0 +1,61 @@
+### Guidelines for Pull Requests and Releases (Web3 1.x)
+
+This document provides some ground rules for contributors (including the maintainer(s) of
+the project) about how to make, review and publish changes to 1.x. The most basic requirement is
+that **Web3 not break**.
+
+## Pull Requests for substantive changes (e.g. everything except comments and docs)
+
+1. Any PR that introduces a logic change should include tests. (In many cases, the tests will take
+more time to write than the actual code).
+
+2. All PRs should sit for 48 hours with the `pleasereview` tag in order to garner feedback.
+
+3. No PR should be merged until it has been reviewed, passes CI, and all reviews' comments are
+addressed.
+
+4. PRs should have a narrow, well-defined focus and make the smallest set of changes possible to achieve their goal. They should include a clear description in the opening comment.
+
+5. Given the choice between a conservative change that mostly works and an adventurous change which
+seems better but introduces uncertainty - prefer the conservative change.
+
+## Reviews
+
+The end-goal of review is to suggest useful improvements to the author. Reviews should finish with
+approval unless there are issues that would result in:
+
+1. Buggy behaviour.
+
+2. Undue maintenance burden.
+
+3. Pessimisation (i.e. speed reduction / meaningful build-size increases).
+
+4. Feature reduction (i.e. it removes some aspect of functionality that users rely on).
+
+5. Avoidable risk (i.e it's difficult to test or hard to anticipate the implications of, without
+being strictly necessary to fix something broken).
+
+## Releases
+
+All releases should be proposed in a PR and subject to community review for a minimum of 72 hours (3 days). 
+
+2. Release review periods should be accompanied by a published `rc` version which can be used for
+sanity checks / additional testing.
+
+3. During release review, the code is frozen unless new changes are proposed, approved and merged.
+Changes reset the release clock and should trigger a new `rc` release.
+
+4. Regular maintainers should manually test the `rc` against a Node project and the published
+minified bundle in a browser context. An external reviewer should verify they've done the same.
+
```

Comment:
```suggestion
5. A release PR does have to be approved at least by two known contributors of the web3.js project

```

Reply:
We should probably also add a point that does define how long an RC release has to be published on NPM until it gets published in the ``latest`` track of npm. One week? @cgewecke

Reply:
@nivida At the moment the `rc` is tied pretty closely to a code freeze and the review window which is 3 days. Do you want to extend that to a week? Or decouple them?

Reply:
Oh, you thought to release the ``rc`` as soon as the release PR is open and to have the ``rc`` published 3d until we move the changes to the ``latest`` track?
I thought we will keep a release PR open for 3d and after do we release a ``rc`` version which does have to be published for a week until we move it to the ``latest`` track. 

I think from a "never doing a bad release again" perspective does the long process with waiting of a week give us some more safety. But on the other hand, does it extremely increase the time until a fix got released. 🤔

Reply:
>  But on the other hand, does it extremely increase the time until a fix got released.

I think we should probably also differentiate between normal releases and ``hot-fix`` releases in this document. In cases of security vulnerabilities or other emergency cases would the goal be I think to have the shortest possible reaction time.

Reply:
> I thought we will keep a release PR open for 3d and after do we release a rc version which does have to be published for a week until we move it to the latest track.

I think having an `rc` available helps with the review. It lets you install the current state in some project and tests things - you aren't restricted to manually reviewing the code which is often difficult to understand all the implications of.

I'm happy to have a longer review window and a week sounds fine to me. Agree that it delays releases and agree we *need a hot-fix protocol*.

Reply:
@cgewecke Great we are on the same page! 💪

Reply:
@nivida. Lengthened the release review window to 1 week in 89704e19f986c36173d7dc3989747afce849236b and just said that if there are changes during the window, people should have 'enough' time to review a new rc that includes them. 

Just lmk if that's not what you meant here.